packages/frontend/src/ee/features/aiCopilot/components/AiAgentPageLayout/AiAgentPageLayout.tsx

```diff
@@ -76,22 +81,34 @@ export const AiAgentPageLayout: React.FC<Props> = ({
         artifactPanelRef.current?.expand();
     };
 
-    const setArtifact = (artifact: ReactNode) => {
-        setContextArtifact(artifact);
+    const setArtifact = (artifact: ReactNode, id: string) => {
```

Comment:
does `setArtifact` need to accept `artifact: ReactNode` as a param? just the id should be enough to decide wether or not artifact is set and then the component would derive rest